src/test/ui/const-generics/eval-privacy.stderr

```diff
@@ -0,0 +1,28 @@
+warning: private type `fn(u8) -> u8 {my_const_fn}` in public interface (error E0446)
+  --> $DIR/eval-privacy.rs:12:1
+   |
+LL | / impl<const U: u8> Trait for Const<U>
+LL | |
+LL | |
+LL | | where
+...  |
+LL | |     }
+LL | | }
+   | |_^
+   |
+   = note: `#[warn(private_in_public)]` on by default
+   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
+   = note: for more information, see issue #34537 <https://github.com/rust-lang/rust/issues/34537>
+
+error[E0446]: private type `fn(u8) -> u8 {my_const_fn}` in public interface
```

Comment:
Would it make sense to change the error msg here?

Reply:
yes it would, do you want to do this in this PR?

We probably also want to always hard error here as there are no future compat issues we have to worry about.

You can either add a `FIXME(const_evaluatable_checked)` or fix this in the current PR as well.

Reply:
ah yea I can fix this in the current PR, what should the type be called?
`const fn`? Or something else

Reply:
hm, not sure how to best handle this.
I guess I would expect "private function `my_const_fn` in public interface".

Reply:
@JulianKnodt 
Could you remove this change (https://github.com/rust-lang/rust/pull/79291#issuecomment-762462438), the message says what it intends to say.

Reply:
Yep, it might take me a bit to get to it though